src/async-wrap.h

```diff
@@ -46,10 +47,16 @@ class AsyncWrap : public BaseObject {
 #undef V
   };
 
-  inline AsyncWrap(Environment* env,
-                   v8::Local<v8::Object> object,
-                   ProviderType provider,
-                   AsyncWrap* parent = nullptr);
+  AsyncWrap(v8::Isolate* isolate,
```

Comment:
If you want to make this header public then you should make all methods non-inline.  Otherwise it may get hard to maintain ABI stability.